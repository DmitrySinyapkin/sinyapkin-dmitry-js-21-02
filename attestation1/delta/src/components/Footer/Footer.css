.footer {
  width: 100%;
  height: 100%;
  font-size: 20px;
  display: flex;
  justify-content: space-between;
  align-items: center;
}
