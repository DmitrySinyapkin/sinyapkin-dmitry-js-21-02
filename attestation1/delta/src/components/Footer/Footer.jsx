import React from "react";
import LanguageChanger from "../LanguageChanger/LanguageChanger";
import ThemeChanger from "../ThemeChanger/ThemeChanger";
import './Footer.css';

const Footer = () => {
    return (
        <div className="footer">
            <span className="footer__copyright">Delta world &#169; 2021</span>
            <div className="footer__changers">
                <LanguageChanger />
                <ThemeChanger />
            </div>
        </div>
    );
}

export default Footer;
