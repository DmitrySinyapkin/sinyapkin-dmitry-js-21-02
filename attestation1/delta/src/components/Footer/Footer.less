@font-size: 20px;

.footer {
    width: 100%;
    height: 100%;
    font-size: @font-size;
    display: flex;
    justify-content: space-between;
    align-items: center;

    .footer__changers {
        display: flex;
        gap: 10px;
        align-items: center;
    }

    @media (max-width: 768px) {
        .footer__copyright {
            display: none;
        }
    }
}
