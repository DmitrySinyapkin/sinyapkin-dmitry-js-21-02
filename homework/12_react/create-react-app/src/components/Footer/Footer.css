.footer {
    padding: 15px 10px;
    display: flex;
    justify-content: space-between;
    background: #f2f2f2;
}

.footer__text {
    color: #666666;
}

.footer__link {
    color: #666666;
}

.footer__link:link, .footer__link:visited {
    text-decoration: none;
}

.footer__link:hover {
    font-weight: bold;
}