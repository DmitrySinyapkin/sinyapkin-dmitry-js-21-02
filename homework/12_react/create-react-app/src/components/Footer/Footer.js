import React from "react";
import './Footer.css';

export class Footer extends React.Component {
    render() {
        return (
            <div className="footer">
                <span className="footer__text"><a href="" className="footer__link">Контакты</a></span>
                <span class="footer__text">&#169; 2021 ИП Рыбов О.А</span>
            </div>
        );
    }
}