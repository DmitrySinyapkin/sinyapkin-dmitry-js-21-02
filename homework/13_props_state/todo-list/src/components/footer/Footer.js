import React from "react";
import './Footer.css'

export class Footer extends React.Component {
    render() {
        return (
            <div className="footer">
                <span>{this.props.copyright}</span>
            </div> 
        );     
    }
}
