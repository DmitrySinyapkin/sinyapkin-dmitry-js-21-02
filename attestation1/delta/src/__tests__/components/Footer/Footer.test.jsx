import React from "react";
import { shallow } from 'enzyme';
import Footer from '../../../components/Footer/Footer';

describe('Header', () => {
    test('snapshot', () => {
        const wrapper = shallow(<Footer />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should render footer children', () => {
        const wrapper = shallow(<Footer />);
        expect(wrapper.find('div.footer').children()).toHaveLength(2);
        expect(wrapper.find('div.footer').find('div.footer__changers').children()).toHaveLength(2);
    });
});
