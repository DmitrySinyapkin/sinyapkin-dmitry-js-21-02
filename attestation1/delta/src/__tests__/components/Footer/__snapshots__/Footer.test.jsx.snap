// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot 1`] = `
<div
  className="footer"
>
  <span
    className="footer__copyright"
  >
    Delta world Â© 2021
  </span>
  <div
    className="footer__changers"
  >
    <LanguageChanger />
    <Connect(ThemeChanger) />
  </div>
</div>
`;
